Update Matrix Documentation for Results (#7012)
tektoncd/pipeline: docs/matrix.md

          value: $(tasks.task-2.results.duh.key) # string replacement from object result
```

### Results from fanned out Matrixed PipelineTasks

> Emitting `Results` from fanned out `PipelineTasks` will not be in the supported in the initial iteration

Reviewer: ```suggestion
Emitting `Results` from fanned out `PipelineTasks` is not currently supported.
```

of `Matrix`. Supporting emitting `Results` from fanned out `PipelineTasks` will be revisited after array

> and object `Results` are supported.

Reviewer: should this sentence be reworded, since array and object results are supported?
